src/poetry/core/masonry/builders/builder.py

```diff
@@ -364,16 +364,6 @@ def convert_script_files(self) -> list[Path]:
 
                 abs_path = Path.joinpath(self._path, source)
 
-                if not abs_path.exists():
```

Comment:
We could put this under `if not self._package.build_script:` to keep the validation. Without a build script I assume the files must exist at this point.

Reply:
In #710 I am doing this, with added test case,  @dimbleby encouraged me to take over. Please review :pray: